pandas-stubs/core/indexes/base.pyi

```diff
@@ -77,12 +80,23 @@ from pandas._typing import (
     np_ndarray_complex,
     np_ndarray_float,
     np_ndarray_str,
+    np_ndarray_td,
     type_t,
 )
 
 class InvalidIndexError(Exception): ...
 
 _ListLike: TypeAlias = ArrayLike | dict[_str, np.ndarray] | SequenceNotStr[S1]
+_NumListLike: TypeAlias = (
+    ExtensionArray
+    | np_ndarray_bool
+    | np_ndarray_anyint
+    | np_ndarray_float
+    | np_ndarray_complex
+    | dict[_str, np.ndarray]
+    | Sequence[complex]
+    | IndexOpsMixin[complex]
+)
```

Comment:
Can we move these 2 types to `core/base.pyi` and import them from there, since they are the same for `Series` and `Index`